### ISSUE TITLE
Client must configure AuthenticationTls even if Authentication using TLS is not enabled

### ISSUE THREAD

**Issue by BewareMyPower** [apache/pulsar#14762]
**Describe the bug**
When I followed the [official tutorial](https://pulsar.apache.org/docs/en/security-tls-transport/) to configure transport encryption using TLS, I found the [example code](https://pulsar.apache.org/docs/en/security-tls-transport/#cli-tools) doesn't work.

It worked after I configured the `AuthenticationTls` at client side, i.e.

```java
        try (PulsarClient client = PulsarClient.builder()
                .serviceUrl("pulsar+ssl://localhost:6651")
                .enableTls(true)
                .tlsTrustCertsFilePath("/path/to/cacert.pem")
                .enableTlsHostnameVerification(false)
                .allowTlsInsecureConnection(false)
                .authentication(AuthenticationTls.class.getName(), authParams) // NOTE: here
                .build()) {
```

However, I only configured the TLS transport encryption, **not authentication**, i.e. I didn't configure `AuthenticationTlsProvider` in `broker.conf`. This behavior is strange and not consistent with the documents.

**To Reproduce**
We can just see the current test for TLS. First, see the setup for Broker:

https://github.com/apache/pulsar/blob/577c2dc8d853d4b74c1c5711179196c2bb7bf86e/bouncy-castle/bcfips-include-test/src/test/java/org/apache/pulsar/client/TlsProducerConsumerBase.java#L64-L76

It doesn't configure the `AuthenticationTlsProvider`.

Then, see the setup for client:

https://github.com/apache/pulsar/blob/577c2dc8d853d4b74c1c5711179196c2bb7bf86e/bouncy-castle/bcfips-include-test/src/test/java/org/apache/pulsar/client/TlsProducerConsumerBase.java#L86-L90

Only if the first argument `addCertificates` is true would `AuthenticationTls` be enabled.

However, see the test code:

https://github.com/apache/pulsar/blob/577c2dc8d853d4b74c1c5711179196c2bb7bf86e/bouncy-castle/bcfips-include-test/src/test/java/org/apache/pulsar/client/TlsProducerConsumerTest.java#L83-L92

When the first argument of `internalSetUpForClient` is false, i.e. `AuthenticationTls` is not configured, the expected behavior is failure.

**Expected behavior**
The client should work even if the authentication for TLS is not enabled if the broker side doesn't configure the `AuthenticationTlsProvider`.

**Comment by BewareMyPower** [comment]
It looks like without Authentication for TLS, we need to generate a cert and a key with any common name.

So I think a solution could be generating a cert and a key with a random generated common name when `tlsTrustCertsFilePath` is configured.

**Comment by BewareMyPower** [comment]
The current document is wrong. Maybe we should update the documents first. Then we might need to make a discussion about if there is a need to generate the cert and key automatically.

**Comment by Technoboy-** [comment]
I'm now working on the authorization part (#14639), its related to authentication. so I will take some time for diving into this.

**Comment by nodece** [comment]
We should split the transport and auth for TLS.

For transport, the code can be so like:
```
PulsarClient client = PulsarClient.builder()
                .enableTls(true)
                .tlsTrustCertsFilePath("ca.pem")
                .tlsCertFilePath("client-ca.pem")
                .tlsKeyFilePath("client-key.pem")
                .build();
```

For auth, the code can be so like:
```
Map<String, String> authParams = new HashMap<>();
authParams.put("tlsCertFile", "client-ca.pem");
authParams.put("tlsKeyFile", "client-key.pem");
PulsarClient client = PulsarClient.builder()
                .enableTls(true)
                .tlsTrustCertsFilePath("ca.pem")
                .authentication(AuthenticationTls.class.getName(), authParams)
                .build();
```

When using the TLS auth, we don't need to set `tlsCertFilePath("client-ca.pem")` and `tlsKeyFilePath("client-key.pem")`, the `authentication` instead of this.


There have an important thing that if we are using the authentication with the token, we cannot setup the TLS transport.

**Comment by nodece** [comment]
I create a [PIP-158](#15289) to fix this.

**Comment by nodece** [comment]
Closed by #15634.
